locals.tf

```diff
@@ -31,47 +30,93 @@ locals {
   kubeconfig_path                       = var.iac_tooling == "docker" ? "/workspace/${local.kubeconfig_filename}" : local.kubeconfig_filename
   kubeconfig_ca_cert                    = data.aws_eks_cluster.cluster.certificate_authority.0.data
 
-  # Mapping node_pools to worker_groups
-  default_node_pool = [
-    {
-      name                                 = "default"
-      instance_type                        = var.default_nodepool_vm_type
-      root_volume_size                     = var.default_nodepool_os_disk_size
-      root_volume_type                     = var.default_nodepool_os_disk_type
-      root_iops                            = var.default_nodepool_os_disk_iops
-      asg_desired_capacity                 = var.default_nodepool_node_count
-      asg_min_size                         = var.default_nodepool_min_nodes
-      asg_max_size                         = var.default_nodepool_max_nodes
-      kubelet_extra_args                   = "--node-labels=${replace(replace(jsonencode(var.default_nodepool_labels), "/[\"\\{\\}]/", ""), ":", "=")} --register-with-taints=${join(",", var.default_nodepool_taints)}"
-      additional_userdata                  = (var.default_nodepool_custom_data != "" ? file(var.default_nodepool_custom_data) : "")
-      metadata_http_endpoint               = var.default_nodepool_metadata_http_endpoint
-      metadata_http_tokens                 = var.default_nodepool_metadata_http_tokens
-      metadata_http_put_response_hop_limit = var.default_nodepool_metadata_http_put_response_hop_limit
-
+  # Mapping node_pools to node_groups
+  default_node_pool = {
+    default = {
+      name                              = "default"
+      instance_types                     = [var.default_nodepool_vm_type]
+      block_device_mappings           = {
+        xvda = {
+          device_name = "/dev/xvda"
+          ebs = {
+            volume_type                     = var.default_nodepool_os_disk_type
+            volume_size                     = var.default_nodepool_os_disk_size
+            iops                            = var.default_nodepool_os_disk_iops
+          }
+        }
+      }
+      desired_size                      = var.default_nodepool_node_count
+      min_size                          = var.default_nodepool_min_nodes
+      max_size                          = var.default_nodepool_max_nodes
+      taints                            = { for i, taint in var.default_nodepool_taints : "default-${i}"=> { 
+                                              "key" = split("=", taint)[0], 
+                                              "value"= split(":", split("=", taint)[1])[0], 
+                                              "effect"=length(regexall(":No", taint)) > 0 ? upper(replace(split(":", split("=", taint)[1])[1], "No", "NO_")) : upper(replace(split(":", split("=", taint)[1])[1], "No", "_NO_"))
```

Comment:
What is this new effect items used for? And why the No filters and adjustments on the taints?

Reply:
See AWS docs https://docs.aws.amazon.com/eks/latest/userguide/node-taints-managed-node-groups.html

Reply:
Might want to add the link to references to these items in the code base so others are aware.

Reply:
https://github.com/sassoftware/viya4-iac-aws/blob/update_eks_module/locals.tf#L93